src/main/groovy/com/deploygate/gradle/plugins/dsl/NamedDeployment.java

```diff
@@ -0,0 +1,211 @@
+        return hasDistribution() ? getDistribution().getKey() : null;
+    }
+
+    @Deprecated
+    public void setDistributionKey(@Nullable final String distributionKey) {
+        DeprecationLogger.deprecation("NamedDeployment.setDistributionKey(String)", "2.0", "3.0", "Use distribution closure instead.");
+        distribution(new Action<Distribution>() {
+            @Override
+            public void execute(Distribution distribution) {
+                distribution.setKey(distributionKey);
+            }
+        });
+    }
+
+    @Deprecated
+    @Nullable
+    public String getReleaseNote() {
+        DeprecationLogger.deprecation("NamedDeployment.getReleaseNote()", "2.0", "3.0", "Use distribution closure directly.");
+        return hasDistribution() ? getDistribution().getReleaseNote() : null;
+    }
+
+    @Deprecated
+    public void setReleaseNote(@Nullable final String releaseNote) {
+        DeprecationLogger.deprecation("NamedDeployment.setReleaseNote(String)", "2.0", "3.0", "Use distribution closure instead.");
+        distribution(new Action<Distribution>() {
+            @Override
+            public void execute(Distribution distribution) {
+                distribution.setReleaseNote(releaseNote);
+            }
+        });
+    }
+
+    @Deprecated
+    public boolean getNoAssemble() {
+        DeprecationLogger.deprecation("NamedDeployment.getNoAssemble()", "2.0", "3.0", "Use isSkipAssemble() instead.");
+        return isSkipAssemble();
+    }
+
+    @Deprecated
+    public void setNoAssemble(boolean noAssemble) {
+        DeprecationLogger.deprecation("NamedDeployment.setNoAssemble(boolean)", "2.0", "3.0", "Use setSkipAssemble() instead.");
+        setSkipAssemble(noAssemble);
+    }
+
+    // end: backward compatibility
+
+    // just for avoiding deprecation logging. non-public api.
+
+    @Deprecated
+    public void _internalSetVisibility(@Nullable String visibility) {
+        this.visibility = visibility;
+    }
+
+    @Deprecated
+    public String _internalGetVisibility() {
+        return visibility;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) return true;
+        if (getClass() != o.getClass()) return false;
+
+        NamedDeployment that = (NamedDeployment) o;
+
+        if (skipAssemble != that.skipAssemble) return false;
+        if (!getDistribution().equals(that.getDistribution())) return false;
+        if (!name.equals(that.name)) return false;
+        if (sourceFile != that.sourceFile) return false;
+        if (!Objects.equals(message, that.message)) return false;
+
+        return true;
+    }
+
+    public int hashCode() {
+        int result;
+        result = name.hashCode();
+        result = 31 * result + (sourceFile != null ? sourceFile.hashCode() : 0);
+        result = 31 * result + (message != null ? message.hashCode() : 0);
+        result = 31 * result + (skipAssemble ? 1 : 0);
+        result = 31 * result + getDistribution().hashCode();
```

Comment:
```suggestion
        result = 31 * result + (hasDistribution() ? getDistribution().hashCode() : 0);
```

Existing code returns 0 if distribution is null.
Is there any reason not to return 0?

https://github.com/DeployGate/gradle-deploygate-plugin/pull/144/files#diff-0ac52e1ba8973482bdd7896222e2135ae8c15a3b8336aadff46afc50683a3614L143

Reply:
These code are auto-generated and following the best practice in JVM world. `null` should be zero, otherwise should have any value.

Reply:
I did not know this code was auto-generated.
Looks good!